test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -563,6 +563,25 @@ def skip_p5_tests(instance_type, processor, ecr_image):
         if processor != "gpu" or Version(image_cuda_version.strip("cu")) < Version("120"):
             pytest.skip("P5 EC2 instance require CUDA 12.0 or higher.")
 
+@pytest.fixture(autouse=True)
+def skip_smdataparallel_p5_tests(instance_type, processor, ecr_image):
+    """SMDDP tests are broken for PyTorch 2.1 on p5 instances, so we should skip
+    """
+    smdataparallel_skip = False
+    skip_dict = {"==2.1": ["cu121"]}
+    if _validate_pytorch_framework_version(
+        request, processor, ecr_image, "skip_smddataparallel_test", skip_dict
+    ):
+        smdataparallel_skip = True
+
+    p5_skip = False
+    if "p5." in instance_type:
+        image_cuda_version = get_cuda_version_from_tag(ecr_image)
+        if processor != "gpu" or Version(image_cuda_version.strip("cu")) < Version("120"):
+            p5_skip = True
+
+    if smdataparallel_skip and p5_skip:
+        pytest.skip("SM Data Parallel tests are not working on P5 instances, skipping test")
 
 def _validate_pytorch_framework_version(request, processor, ecr_image, test_name, skip_dict):
```

Comment:
While this function is named _validate_pytorch*, it appears that it does not actually check whether or not the framework version of the image is pytorch. Especially because these are autouse, we should replace "_" with "framework" and do a check to ensure we are looking at a pytorch image

Reply:
I think I've addressed this in the latest updates.

Reply:
It's not addressed in this revision ([5365a93](https://github.com/aws/deep-learning-containers/pull/3730/commits/5365a93344c1a86eea7cfc66519573d40f3dfe0f)), but cc @sirutBuasai who will be opening a separate PR to handle this

Reply:
@dkey-amazon has this been addressed? 

I think the method `_validate_pytorch_framework_version` had been meant for `pytorch`itself. However, as @arjkesh mentioned, looks like there is no explicit check for this.

If @sirutBuasai is handling this at a broader perspective, I think we can get this PR merged, using the `_validate_pytorch_framework_version` method in its old form.